tests/test_gen.py

```diff
@@ -122,3 +122,10 @@ def test_cant_pin():
     with pytest.raises(Exception) as cm:
         jdist.pinned
     assert str(cm.value) == "Can not pin because no version available is in spec"
+
+
+def test_whl_extras():
+    whl_fname = os.path.join(here, "testwhlextra-1.0.0-py3-none-any.whl")
+    print(f'whl_fname={whl_fname}')
```

Comment:
This cause CI failure (we cannot use f'string' because johnnydep still supports Python 2.7)